lib/metasploit/framework/compiler/mingw.rb

```diff
@@ -59,15 +60,54 @@ def build_cmd(src)
           cmd
         end
 
-        def cleanup_files
-          src_file = "#{self.file_name}.c"
-          exe_file = "#{self.file_name}.exe"
 
+        def compile_cpp(src)
+          cmd = build_cpp_cmd(src)
+          print("#{cmd}\n") if show_compile_cmd
+          stdin_err, status = Open3.capture2e(cmd)
+          stdin_err
+        end
+
+        def compile_cpp_files(files)
+          cmd = build_cpp_files_cmd(files)
+          print("#{cmd}\n") if show_compile_cmd
+          stdin_err, status = Open3.capture2e(cmd)
+          stdin_err
+        end
+
+        def build_cpp_cmd(src)
+          src_file = Tempfile.create(['scexec', ".cpp"])
+          File.write(src_file.path, src)
+          build_cpp_files(src_file.path)
+        end
+
+        def build_cpp_files_cmd(file_array)
+          cmd = [ mingw_bin ]
+          cmd << file_array.join(" ")
+          cmd << "-I #{INCLUDE_DIR}"
+          cmd << "-o #{outfile}"
+          cmd << "-shared" if outfile.ends_with?('.dll')
+
+          cmd << opt_lvl if OPTIMIZATION_FLAGS.include?(opt_lvl)
+          cmd << compile_options
+
+          link_opts = ['-Wl']
+          link_opts << '-s' if strip_syms
+          link_opts << link_options
+          link_opts << "-T#{link_script} " if link_script
```

Comment:
According to https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html I think there is meant to be a space between `-T` and the option being linked?

The other option would be to do `-T` as one entry in the array, and the value as another entry in the array. Then when you combined everything together separated by commas it should work.

```
Pass option as an option to the linker. If option contains commas, it is split into multiple options at the commas. You can use this syntax to pass an argument to the option. For example, -Wl,-Map,output.map passes -Map output.map to the linker. When using the GNU linker, you can also get the same effect with -Wl,-Map=output.map.
```

So this would follow that pattern and would prevent you having to use something that might be specific to the GNU linker supporting `=` signs.

Reply:
You're right. Nice catch. I've opted for keeping individual flags and their opts as individual array elements and just splatting them into Open3